clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -67,84 +67,41 @@ static constexpr llvm::StringLiteral LibstdcxxFailedAssertion =
     "__failed_assertion";
 constexpr unsigned MaxKernelArgsSize = 2048;
 
-namespace {
+static bool isSyclType(QualType Ty, SYCLTypeAttr::SYCLType TypeName) {
```

Comment:
I wonder, why not anonymous namespace, but a `static` function?

Reply:
I didn't have any reason. I deleted the namespace along with the Util class which it held when I first started working on this and just added static functions as I needed. I can change this back to anonymous namespace if you prefer.

Reply:
I don't have a strong position here, so I will leave it up to you (or FE folks) to decide

Reply:
clang guidelines have this to say:
https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

"The problem with anonymous namespaces is that they naturally want to encourage indentation of their body, and they reduce locality of reference: if you see a random function definition in a C++ file, it is easy to see if it is marked static, but seeing if it is in an anonymous namespace requires scanning a big chunk of the file."

and:

"Avoid putting declarations other than classes into anonymous namespaces:"

Reply:
Thank you Prem!